### PR TITLE
[Spec] Add text on handling bookmarks and related

### LINKED ISSUE #150: Clarify URL fragment stripping cases in spec
The [spec discusses](https://wicg.github.io/scroll-to-text-fragment/#parsing-the-fragment-directive) how the text-fragment portion of a URL fragment is stripped out during loading.

As noted in https://lists.webkit.org/pipermail/webkit-dev/2020-November/031572.html, there's various edge cases like script and bookmarks where we should explicitly clarify the behavior.

### PR DESCRIPTION
**Pull request by bokand**
Adds some non-normative text on how UAs could/should behave with respect
to features like the location bar, bookmarks, sharing, etc.

Also adds some language that UAs should consider how to differentiate
the visual indicator from page content, as mentioned in #118.

Fixes #150

***
<a href="https://pr-preview.s3.amazonaws.com/bokand/ScrollToTextFragment/pull/155.html" title="Last updated on Nov 16, 2020, 11:06 PM UTC (e55dada)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/scroll-to-text-fragment/155/e2318d2...bokand:e55dada.html" title="Last updated on Nov 16, 2020, 11:06 PM UTC (e55dada)">Diff</a>

### CODE REVIEW

**Comment by bokand** [comment]
@nickburris ptal

File: index.bs [hunk 1]
```diff
@@ -1628,6 +1628,107 @@ exfiltration.
...
+#### Bookmarks #### {#urls-in-bookmarks}
+
+Many UAs provide a "bookmark" feature allowing users to store a convenient link
+to the current page in the UA's interface.
+
+A newly created bookmark should, by default, include the [=fragment directive=]
+in the URL if, and only if, a match was found and the visual indicator hasn't
+been dismissed.
+
+Navigating to a URL from a bookmarks should process a [=fragment directive=] as
```
Comment (nickburris) [inline index.bs:1720]: "bookmark~s~"

**Review by nickburris — APPROVED** [review]
LGTM, this looks great! Just a couple nits

File: index.bs (hunk 1 above)
Comment (bokand) [reply index.bs]: Done

File: index.bs [hunk 2]
```diff
@@ -1628,6 +1628,107 @@ exfiltration.
...
+  experience. E.g. perhaps a URL should elide the text fragment if the user
+  scrolls it out of view?
+  </p>
+</div>
+
+The general principle is that a URL should include the [=fragment directive=]
+only while the visual indicator is visible (i.e. not dismissed). If the user
+dismisses the indicator, the URL should not include the [=fragment directive=].
+
+If the URL includes a text fragment but a match wasn't found on in the current
```
[the remark this answers is not in the document]
Comment (bokand) [reply index.bs]: Done

**Comment by bokand** [comment]
Thanks for the quick review!

**Pull request merged**